src/pages/dashboardTab/store/dashboard-temp-reducer.js

```diff
@@ -0,0 +1,1440 @@
+  if (!hasFilter()) {
+    return selectedRowInfo.length > 0;
+  }
+
+  const filteredNames = await getFileNamesByFileIds(getState().filteredFileIds);
+  switch (getState().currentActiveTab) {
+    case tabIndex[2].title:
+      filteredIds = filteredNames;
+      selectedRowInfo = getState().dataFileSelected.selectedRowInfo;
+
+      break;
+    case tabIndex[1].title:
+      filteredIds = getState().filteredSampleIds;
+      selectedRowInfo = getState().dataSampleSelected.selectedRowInfo;
+      break;
+    default:
+      filteredIds = getState().filteredSubjectIds;
+      selectedRowInfo = getState().dataCaseSelected.selectedRowInfo;
+  }
+
+  return selectedRowInfo.filter(
+    (value) => (filteredIds && filteredIds !== null ? filteredIds.includes(value) : false),
+  ).length > 0;
+}
+
+function setDataCaseSelected(result) {
+  store.dispatch({ type: 'SET_CASES_SELECTION', payload: result });
+}
+
+function setDataFileSelected(result) {
+  store.dispatch({ type: 'SET_FILE_SELECTION', payload: result });
+}
+
+function setDataSampleSelected(result) {
+  store.dispatch({ type: 'SET_SAMPLE_SELECTION', payload: result });
+}
+/**
+ *  Returns the functuion depend on current active tab
+ * @return {func}
+ */
+
+export function getTableRowSelectionEvent() {
+  const currentState = getState();
+  const tableRowSelectionEvent = currentState.currentActiveTab === tabIndex[2].title
+    ? setDataFileSelected
+    : currentState.currentActiveTab === tabIndex[1].title
+      ? setDataSampleSelected : setDataCaseSelected;
+  return tableRowSelectionEvent;
+}
+
+export function clearTableSelections() {
+  store.dispatch({ type: 'CLEAR_TABLE_SELECTION' });
+}
+
+export function setCaseSelected(payload) {
+  store.dispatch({ type: 'SET_CASE_SELECTION', payload });
+  if (!payload) {
+    store.dispatch(fetchDashboardTab());
+  }
+}
+
+export const getDashboard = () => getState();
+
+const getFilterKeys = (originalObj) => {
+  const filterObjKeys = Object.keys(originalObj).filter((key) => key.includes('filter')
+    || key.includes('subject')
+    || key.includes('arms')
+    || key.includes('nodeCount'));
+  const newObj = {};
+  filterObjKeys.forEach((key) => {
+    newObj[key] = originalObj[key];
+  });
+  return newObj;
+};
+
+// reducers
+const reducers = {
+  DASHBOARDTAB_QUERY_ERR: (state, item) => ({
+    ...state,
+    hasError: true,
+    error: item,
+    isLoading: false,
+    isFetched: false,
+  }),
+  READY_DASHBOARDTAB: (state) => ({
+    ...state,
+    isLoading: false,
+    isFetched: true,
+    setSideBarLoading: false,
+    isDashboardTableLoading: false,
+  }),
+  TOGGGLE_CHECKBOX_WITH_API: (state, item) => {
+    // get all the filter objects from the raw data.
+    const filterCheckbox = getFilterKeys(state.rawApiData);
+    console.log('DEBUG_REAW_API', filterCheckbox);
+    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(
+      filterCheckbox, facetSearchData,
+    );
```

Comment:
Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502144)
```suggestion
    const checkboxData1 = setSelectedFilterValues(updatedCheckboxData1, item.allFilters);
```